packages/migrations/src/actions/2024.09.26T00.00.00.create-preflight-scripts.ts

```diff
@@ -0,0 +1,22 @@
+import { type MigrationExecutor } from '../pg-migrator';
+
+export default {
+  name: '2024.09.26T00.00.00.create-preflight-scripts.sql',
+  run: ({ sql }) => sql`
+CREATE TABLE "document_preflight_scripts" (
+  "id" uuid NOT NULL DEFAULT uuid_generate_v4(),
+  "source_code" text NOT NULL,
+  "target_id" uuid NOT NULL REFERENCES "targets"("id") ON DELETE CASCADE,
+  "created_by_user_id" uuid REFERENCES "users"("id") ON DELETE SET NULL,
+  "created_at" timestamptz NOT NULL DEFAULT now(),
+  "updated_at" timestamptz NOT NULL DEFAULT now(),
+  PRIMARY KEY ("id")
+);
+
+CREATE INDEX "document_preflight_scripts_connection_pagination" ON "document_preflight_scripts" (
+  "target_id" ASC,
+  "created_at" DESC,
+  "id" DESC
+);
```

Comment:
We only have one preflight script per organization, so we don't need pagination for it.

We just need an index on "target_id" to quickly find the preflight script for a given target.

Also we should avoid the creation of multiple preflight scripts for the same target. We can set a unique constraint on the `"target_id"` field to avoid duplicate records for the same target on the db level. But we should also have that being handled within our business logic.